pkg/kubelet/pod/pod_manager.go

```diff
@@ -241,6 +248,16 @@ func (pm *basicManager) DeletePod(pod *v1.Pod) {
 		delete(pm.mirrorPodByFullName, podFullName)
 		delete(pm.translationByUID, mirrorPodUID)
 	} else {
+		var gracePeriod int64
+		if pod.Spec.TerminationGracePeriodSeconds != nil {
+			gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
+			klog.V(3).Infof("using grace period %v from %q", gracePeriod, podFullName)
+		}
+		pm.pendingPodByFullName[podFullName] = kubetypes.UIDWithGracePeriod{
```

Comment:
this map is populated for every pod, but it appears to only be cleaned up when calling `DeletePendingPod`, which appears to only be called when deleting mirror pods.  is this leaking entries for non mirror pods, or am I missing something?

Reply:
Added check for static pod.